### ISSUE TITLE
Unable to open localhost:6666 in browsers due to restricted port 6666

### ISSUE THREAD

**Issue by mohanraj-r** [pact-foundation/pact-go#63]
After running the daemon, I opened http://localhost:6666 in local web-browsers as a quick check to see if the daemon was running and curious to see the default http response if any. But none of major web-browsers (Chrome, Firefox, Safari) were able to open http://localhost:6666. This misled me into thinking that the daemon was probably not running and I ended up spending time changing the ports in the daemon and the example src to verify the tests running. 

Just found that 6666 is [one of restricted ports](https://superuser.com/a/188070) and hence blocked. And the error messages from the browsers are not very helpful in diagnosing this issue (esp. Chrome). Think it might be better to switch to a unrestricted port to prevent confusion like this among beginners. 

## Software versions
* Browsers
  - Chrome 63.0.3239.132
  - Firefox 57.0.4
  - Safari 11.0.2
* **OS**: Mac OSX 10.13.2 
* **Consumer Pact library**: 
```
$ pact-go version
Pact Go CLI v0.0.11
```
* **Provider Pact library**: [v0.0.12](https://github.com/pact-foundation/pact-go/releases/download/v0.0.12/pact-go_darwin_amd64.tar.gz)
* **Golang Version**: `1.9.2`
* **Golang environment**: 
```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/mohanrajr/go"
GORACE=""
GOROOT="/usr/local/Cellar/go/1.9.2/libexec"
GOTOOLDIR="/usr/local/Cellar/go/1.9.2/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/d0/742yxrrx0glg93s4z_rwyzkxm0whbv/T/go-build176059632=/tmp/go-build -gno-record-gcc-switches -fno-common"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
```

## Expected behaviour
http://localhost:6666 opens successfully in a web-browser 

## Actual behaviour
Error messages when opening http://localhost:6666
Chrome : 
```
This site can’t be reached
The webpage at http://localhost:6666/ might be temporarily down or it may have moved permanently to a new web address.
ERR_UNSAFE_PORT
```

Firefox:
```
This address is restricted

This address uses a network port which is normally used for purposes other than Web browsing. Firefox has canceled the request for your protection.
```

## Steps to reproduce
* Run the daemon `./pact-go daemon -l DEBUG -v`
* Open http://localhost:6666/ in a web-browser

## Relevant log files
Nothing relevant in the log files

**Comment by mohanraj-r** [comment]
On a related note I think it will be useful to use an env var in the src for the ports instead of hard-coding 6666. The daemon could set the env var when it starts up.

**Comment by mefellows** [comment]
@mohanraj-r couple of things here.

1) The daemon is not a browsable thing, it is an RPC endpoint - so a browser like Chrome will not be able to talk to it
2) The daemon is configurable, you can pass the port in on the CLI:

```
λ ./pact-go daemon --help
Creates a daemon for the Pact DSLs to communicate with

Usage:
  pact-go daemon [flags]

Flags:
  -a, --address string   Local network address to listen on (e.g. '', '127.0.0.1', '[::1]' etc.)
  -n, --network string   Local network interface to listen on ('tcp', 'tcp4', 'tcp6') (default "tcp")
  -p, --port int         Local daemon port to listen on (default 6666)

Global Flags:
  -l, --logLevel string   Set the logging level (DEBUG, INFO, ERROR) (default "INFO")
  -v, --verbose           verbose output (default true)
```

**Comment by mohanraj-r** [comment]
Thanks @mefellows 

> The daemon is not a browsable thing, it is an RPC endpoint that is used by the framework - so a browser like Chrome will not be able to talk to it

Yes, I understand that. It was just a quick check way to determine if the service was up. Not an ideal way to do it I understand.

> The daemon - including the port - is configurable, you can pass the port in on the CLI:

Yes, I did end up using the `--port` option. But then the issue is that `6666` is hardcoded in the tests and I had to edit the test src to match the `--port` option for the tests to work. It is here my suggestion to have them read off an env var instead of using the hardcoded value.
